File: Backend/src/main/java/com/example/demo/entity/Flower.java
package com.example.demo.entity;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Entity
@RequiredArgsConstructor
@AllArgsConstructor
@Data
public class Flower {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	
	@Column
	private String name;
	
	@Column
	private double price;
	
	@Column
	private String imageURL;
	@Column
	private int quantity;
	
    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private FlowerCategory category;

}

File: frontend/src/app/common/flower.ts
import { FlowerCategory } from './flower-category';

export class Flower {
    id: number ;
	name: string ;
	price: number ;
    imageURL: String ;
    quantity:number;
    category: FlowerCategory;

    constructor(flower:Flower){
        this.id=flower.id;
        this.name=flower.name;
        this.imageURL=flower.imageURL;
        this.price=flower.price;

        this.quantity=1;
    }

}
